src/com/rcv/CastVoteRecord.java

```diff
@@ -1,23 +1,50 @@
 package com.rcv;
 
-import com.fasterxml.jackson.annotation.JsonAnySetter;
 
 import java.util.*;
 
-// map of contest IDs to ContestRankings objects
-// i.e. a Ballot
+// internal representation of a cast vote record
+// contains an ID
+// contains a mapping of rank (integer) to a set of candidate ID(s) selected at that rank
+// a set is used to handle overvotes
+
 public class CastVoteRecord {
-  private Map<Integer, ContestRankings> rankings = new HashMap<Integer, ContestRankings>();
+  public String mBallotID;
+  List<ContestRanking> mRankings;
+  SortedMap<Integer, Set<String>> mSortedRankings;
 
-  @JsonAnySetter
-  public void add(String key, ContestRankings value) {
-    Integer electionId = Integer.parseInt(key);
-    rankings.put(electionId, value);
-  }
+  // mDescriptionsByRound contains who this ballot counted for in each round
+  // followed by reason for exhaustion if it is ever exhausted
+  public List<String> mDescriptionsByRound = new ArrayList<>();
 
-  public CastVoteRecord() {}
+  // adds the string to this CVR round by round descriptions for auditing
+  public void addRoundDescription(String description) {
+    mDescriptionsByRound.add(description);
+  }
+  
+  // output is our rankings sorted from high to low
```

Comment:
You should clarify what you mean by "high to low", since that's ambiguous when you're talking about rankings. Maybe "more preferred to less preferred"?

Reply:
done